backend/src/main/java/heartbeat/service/source/github/GithubService.java

```diff
@@ -16,11 +21,44 @@ public class GithubService {
 
 	public GithubResponse verifyToken(String githubToken) {
 		String token = "token " + githubToken;
-		final var githubRepos = githubFeignClient.getAllRepos(token);
+		log.info("[Github] Start to query repository_url by token: https://api.github.com/user/repos");
+		final var githubReposByUser = githubFeignClient.getAllRepos(token)
+			.stream()
+			.map(GithubRepos::getHtml_url)
+			.toList();
+		log.info("[Github] Successfully queried repository_data by token");
+		log.info("[Github] Start to query organization_url by token: https://api.github.com/user/orgs");
+
+		final var githubOrganizations = githubFeignClient.getGithubOrgsInfo(token);
+		log.info("[Github] Successfully queried organization_data by token");
+
+		List<String> githubRepos = new ArrayList<>(githubReposByUser);
+		log.info(
+				"[Github] Start to query repository_url by organization_name and token: https://api.github.com/orgs/{organization-name}/repos");
+
+		getGithubReposByOrganizations(token, githubOrganizations, githubRepos);
+		log.info("[Github] Successfully queried repository_data by organizations and token");
+		log.info("[Github] Start to remove duplicate githubRepos list");
+		removeDuplicates(githubRepos);
+		log.info("[Github] Successfully remove duplicate githubRepos list");
+
+		return GithubResponse.builder().githubRepos(githubRepos).build();
+	}
+
+	private void getGithubReposByOrganizations(String token, List<GithubOrgsInfo> githubOrganizations,
+			List<String> githubRepos) {
+		githubRepos.addAll(githubOrganizations.stream()
+			.map(GithubOrgsInfo::getLogin)
+			.flatMap(org -> githubFeignClient.getReposByOrganizationName(org, token)
+				.stream()
+				.map(GithubRepos::getHtml_url))
+			.toList());
+	}
 
-		return GithubResponse.builder()
-			.githubRepos(githubRepos.stream().map(GithubRepos::getHtml_url).toList())
-			.build();
+	private static void removeDuplicates(List<String> githubRepos) {
```

Comment:
I am not sure whether it is necessary to have this function to remove duplicates. Maybe you could use `.distinct()` in the stream or `.collect(Collectors.toSet())` directly.

Reply:
have already removed removeDuplicates method and change response from List to LinkedHashSet

Reply:
done